src/host.rs

```diff
@@ -1,14 +1,23 @@
 use crate::envelope::DeliveryInstructions;
+use crate::net::{Segment, SocketPair, StreamEnvelope, Syn};
 use crate::{version, Envelope, Message};
 
 use indexmap::IndexMap;
 use std::collections::VecDeque;
+use std::io;
 use std::net::SocketAddr;
 use std::rc::Rc;
+use std::sync::Arc;
 use tokio::sync::Notify;
 use tokio::time::{Duration, Instant};
 
-/// A host in the simulated network
+/// A host in the simulated network.
+///
+/// Hosts support two networking modes:
+/// - Datagram via [`crate::io::send`] and [`crate::io::recv`]
+/// - Stream via [`crate::net::Listener`] and [`crate::net::Stream`]
+///
+/// Both modes may be used by host software simulatanesouly.
```

Comment:
Lots of duplication here, but it's all internal. I'd like to get the public part and tests in and we can clean this up.